inferno/dataset.py

```diff
@@ -72,6 +79,15 @@ def multi_indexing(data, i):
     2  3  6
 
     """
+    if isinstance(i, np.ndarray):
+        if i.dtype == bool:
+            i = i.nonzero()[0]
```

Comment:
Because the output is a tuple of numpy arrays but pytorch wants a list of integers as index.

Technically, when the tuple contains more than one list, pytorch will complain anyway, so we could get away with only casting the first element to a list, but that would be kind of a silly optimization. And if pytorch one day allows a tuple of lists, our code does not need to be adjusted.